Mixin/UserInterface/Controllers/Home/Apps/Cells/AppFolderCell.swift

```diff
@@ -0,0 +1,141 @@
+                }
+                break
+            }
+        }
+    }
+    
+    func move(to page: Int, animated: Bool) {
+        guard page < apps.count else { return }
+        let indexPath = IndexPath(item: page, section: 0)
+        collectionView.scrollToItem(at: indexPath, at: .left, animated: animated)
+    }
+    
+    func move(view: UIView, toCellPositionAtIndex index: Int, completion: (() -> Void)? = nil) {
+        guard let currentPageCell = collectionView.cellForItem(at: IndexPath(item: currentPage, section: 0)) as? AppPageCell,
+              let flowLayout = currentPageCell.collectionView.collectionViewLayout as? UICollectionViewFlowLayout,
+              let layoutAttributes = flowLayout.layoutAttributesForItem(at: IndexPath(item: index, section: 0)) else {
+            return
+        }
+        if layoutAttributes.frame.minX == 0 {
+            layoutAttributes.frame.origin.x = flowLayout.sectionInset.left
+        }
+        let convertedRect1 = convert(layoutAttributes.frame, from: currentPageCell)
+        let convertedRect2 = convert(convertedRect1, to: view.superview!)
+        UIView.animate(withDuration: 0.35, animations: {
+            view.frame = convertedRect2
+        }, completion: { _ in
+            completion?()
+        })
+    }
+    
+    func revokeFolderCreation(completion: @escaping () -> Void) {
+        guard let currentPageCell = collectionView.cellForItem(at: IndexPath(item: currentPage, section: 0)) as? AppPageCell,
+              let appCell = currentPageCell.collectionView.cellForItem(at: IndexPath(item: 0, section: 0)) as? AppCell else {
+            return
+        }
+        let convertedRect1 = currentPageCell.convert(appCell.imageView?.frame ?? .zero, from: appCell)
+        let convertedRect2 = convert(convertedRect1, from: currentPageCell)
+        let imageView = UIImageView(frame: convertedRect2)
+        imageView.image = appCell.imageView?.image
+        contentView.addSubview(imageView)
+        appCell.imageView?.isHidden = true
+        UIView.animate(withDuration: 0.55, animations: {
+            imageView.transform = .transform(rect: imageView.frame, to: self.imageContainerView.frame)
+            self.imageContainerView.transform = CGAffineTransform.identity.scaledBy(x: 0.01, y: 0.01)
+            self.label?.alpha = 0
+        }, completion: { _ in
+            self.placeholderView = imageView
+            appCell.imageView?.isHidden = false
+            completion()
+        })
+    }
+    
+}
+
+extension AppFolderCell: UICollectionViewDataSource, UICollectionViewDelegate {
+    
+    func collectionView(_ collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
+        return apps.count
+    }
+    
+    func collectionView(_ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
+        guard indexPath.item < apps.count else {
+            return UICollectionViewCell()
```

Comment:
Do not initialize and return a UICollectionView here directly. It will crash the app. If you rather the app to crash here, use `fatalError`; Or if you believe this will never be executed, just dequeue a reusable cell normally and don't put any content to it.